metatron-app/metatron-discovery
# Password maximum usage period, minimum usage period applied.

### Description
Password maximum usage period, minimum usage period applied.

1. Users cannot log in after the maximum password usage period and must change the password through the password change dialog.

2. Users cannot change it again within the minimum period of use after changing the password. (but Administrator can change the password.)

**Related Issue** : 

### How Has This Been Tested?
1. add password period config to application.yaml
For testing, the minimum password usage period is 5 minutes, and the required change period is 10 minutes.
```
polaris:
  user:
    password:
      requiredChangePeriod: PT10M           # The password must be changed within that period. (ISO 8601 Duration Format)
      minimumUsePeriod: PT5M               # The password cannot be changed again within that period. (ISO 8601 Duration Format)
```
2. register userA
3. approve user using admin.
4. logout and wait 10 minutes.
5. login with userA
6. see change password dialog.
7. login with userA.
8. change password in 5 minutes.
9. see error. ("you cannot change password")

#### Need additional checks?


### Types of changes



### Checklist:


### Additional Context
